### ISSUE TITLE
Allow Self-Signed Certificate

### ISSUE THREAD

**Issue by NGC602** [BLeeEZ/amperfy#67]
There is currently no initial setting upon launching the app to skip Certificate Validation for self hosted servers with self signed certificates not authorized by Let’s Encrypt or other. 
Is it possible to implement this feature so it can support LAN only (or over VPN) network configurations?

Thanks for your time

**Comment by NGC602** [comment]
Issue persists even after trusting the certificate to the device being used to connect
![image](https://user-images.githubusercontent.com/84300098/129789178-8df70248-3185-48b9-b021-956784bba168.jpeg)

**Comment by BLeeEZ** [comment]
Interesting. I thought after trusting the certificate and adding full trust the connection should be possible. There was already in the past a similar issue: https://github.com/BLeeEZ/amperfy/issues/18
I am currently in vacation. I will take a look at this issue and try get further information.

**Comment by BLeeEZ** [comment]
In the meantime I read further about this topic and I am not sure how to fix it.
All I can find is to trust as root certificate (what you already did) and then it should work.
The error message you are getting displayed comes directly from the apple request manager (which handles all the https security mechanisms).

**Comment by RedS6X** [comment]
Hey @BLeeEZ  Finally I got more time and I found the way to fit with [Apple requirements](https://support.apple.com/en-us/HT210176). 
This worked for me (Maybe someone will need it 🙏): 
`openssl req -x509 -nodes -newkey rsa:2048 -keyout local.key -out local.crt -days 825 \
-subj '/CN=Apple/C=UK/OU=Apple/O=Apple' \
-addext "keyUsage = digitalSignature, keyEncipherment, dataEncipherment, cRLSign, keyCertSign" \
-addext "extendedKeyUsage = serverAuth, clientAuth, 1.3.6.1.4.1.311.80.1" \
-addext "subjectAltName=DNS:Real Server DNS,IP:Real Server IP"`

After this I got an "401" error and the stars aligned 😆, then I notice, you have updated the app with "Legacy Subsonic". So now everything works like a charm!

PS: Would be nice to get songs info/album image from ID3 Tags.


Best regards,
Thank you again the app is Great!
